CMakeLists.txt

```diff
@@ -0,0 +1,46 @@
+
+cmake_minimum_required(VERSION 3.20)
```

Comment:
not sure what the minimum we can support is I just picked 3.20 randomly

Reply:
I have some tweaks I can make in this regard, also general CMake stuff that I do, but generally this `cmake_minimum_required(VERSION 3.15...3.31)`.